src/volume_mgr.rs

```diff
@@ -775,6 +775,13 @@ where
 
     /// Close a file with the given full path.
     pub fn close_file(&mut self, file: RawFile) -> Result<(), Error<D::Error>> {
+        let file_idx = self.flush_file(file)?;
+        self.open_files.swap_remove(file_idx);
+        Ok(())
+    }
+
+    /// Flush (update the entry) for a file with the given full path.
```

Comment:
change "given full path" to "given raw file handle".

Reply:
It also doesn't make sense to return the internal index here. Maybe make this a private function, and have the public `close_file` and `flush_file` call it?